packages/@lwc/engine-dom/src/patches/global-registry.ts

```diff
@@ -369,18 +397,25 @@ export function patchCustomElementRegistry() {
     // @ts-ignore
     window.HTMLElement = HTMLElement;
 
-    // this method patches the dom and returns helper function for LWC to register names and associate
-    // them to a constructor while returning the pivot constructor ready to instantiate via `new`
-    return (tagName: string, constructor: CustomElementConstructor): CustomElementConstructor => {
+    // This method patches the dom and returns a helper function for LWC to register names and associate
+    // them to a constructor while returning the pivot constructor, ready to instantiate via `new`.
+    return function definePivotCustomElement(
+        tagName: string,
+        constructor: CustomElementConstructor
+    ): CustomElementConstructor {
         tagName = tagName.toLowerCase();
         let PivotCtor = pivotCtorByTag.get(tagName);
         if (isUndefined(PivotCtor)) {
             const definition = getDefinitionForConstructor(constructor);
             PivotCtor = createPivotingClass(tagName, definition);
             definition.PivotCtor = PivotCtor;
+            // We keep a local definitions by tag to keep track separately of our own custom elements
+            // compared to global ones.
+            // The reason we don't need a `localDefinitionsByClass` is because we don't care about duplicate
+            // classes – we will never call `definePivotCustomElement` twice with the same class
+            // but different tags.
             localDefinitionsByTag.set(tagName, definition);
```

Comment:
I wonder about this line. If this line is only used when there is no pivot for a name... because the pivot can be created via a global registration via define() as well. Looking at the code, `localDefinitionsByTag` is only used on `whenDefined`, in the following block:

```
            const localDefinition = localDefinitionsByTag.get(tagName);
            if (!isUndefined(localDefinition) && !isUndefined(localDefinition.UserCtor)) {
                return localDefinition.UserCtor;
            }
```

that is wrong, because a local definition should never leak to user-land via the global registry. so in principle, I missed up there. now, what happen when you are expecting for something to be defined, but it is lwc who claims it? I believe the idea there is then to await until a global definition is added via `customElements.define()`. I thought I have implemented that, maybe I was wrong there too.

Reply:
Look at what I"m doing here: https://github.com/caridy/redefine-custom-elements/blob/main/src/index.ts#L441-L446, and that promise can only be resolved when a qualifying define() is invoked.

Reply:
OK, I hadn't considered that the LWC Pivot constructor should not leak to `customElements.whenDefined()` or `customElements.get()`. I'll have to look into that.

I got rid of `definedPromises` and `definedResolvers` because I could not find any test code that would actually hit those lines of code. Can you think of a way to invoke them?